core/src/util/builders.cpp

```diff
@@ -383,7 +372,7 @@ void Builders::buildPolyLine(const Line& _line, PolyLineBuilder& _ctx) {
         for (size_t i = 0; i < _line.size() - 1; i++) {
             const glm::vec3& coordCurr = _line[i];
             const glm::vec3& coordNext = _line[i+1];
-            if (isOnTileEdge(coordCurr, coordNext)) {
+            if (isOutsideTile(coordCurr, coordNext)) {
                 Line line = Line(&_line[cut], &_line[i+1]);
                 buildPolyLineSegment(line, _ctx);
```

Comment:
The good old cohen-sutherland trivial reject case :)

I would add the tolerance to slightly extend the tile bbox. By subtracting there is a chance that a real boundary line gets omitted